api/include/opentelemetry/plugin/tracer.h

```diff
@@ -39,7 +39,7 @@ class Span final : public trace::Span
 
   void UpdateName(nostd::string_view name) noexcept override { span_->UpdateName(name); }
 
-  void End() noexcept override { span_->End(); }
+  void End(const trace::EndSpanOptions &options = {}) noexcept override { span_->End(options); }
```

Comment:
Is there a reason we have different structures for StartSpanOptions and EndSpanOptions?

Would it make sense to have one common options class, such as **SpanOptions** ?

Reply:
The intent behind this design is to make it very clear to the API user which option can be set at which time (at start and end).

This also reflects the OpenTracing API, which provides [FinishSpanOptions](https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/span.h#L73) and [StartSpanOptions](https://github.com/opentracing/opentracing-cpp/blob/master/include/opentracing/tracer.h#L25).